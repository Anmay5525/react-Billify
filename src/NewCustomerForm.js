const { useRef } = React;

export default function NewCustomerForm({ handleSubRouteChange }) {
  const name = useRef();
  const email = useRef();
  const phone = useRef();

  const handleNewCustomer = (e, name, email, phone) => {
    e.preventDefault();
    const url = "https://rzp-training.herokuapp.com/team1/customers";

    const data = {
      name,
      email,
      contact: phone,
    };

    fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    })
      .then((res) => {
        if (res.ok) {
          return res.json();
        } else {
          alert(
            "Something went wrong. Server responded with status " + res.status
          );
        }
      })
      .then((r) => {
        if (r) {
          alert("New customer created");
          handleSubRouteChange("list");
        }
      })
      .catch((error) => console.log(error));
  };

  return (
    <form
      className="customers-form"
      onSubmit={(e) =>
        handleNewCustomer(
          e,
          name.current.value,
          email.current.value,
          phone.current.value
        )
      }
    >
      <div className="input-cnt">
        <div>Name</div>
        <input
          ref={name}
          className="input"
          id="name"
          type="text"
          placeholder="Enter name of customer"
        />
      </div>

      <div className="input-cnt">
        <div>Email</div>
        <input
          ref={email}
          className="input"
          type="text"
          id="email"
          placeholder="Enter email"
        />
      </div>

      <div className="input-cnt">
        <div>Phone No.</div>
        <input
          ref={phone}
          className="input"
          type="number"
          id="phone"
          placeholder="Enter phone no."
        />
      </div>
      <div>
        <button className="customers-add-btn">Add Customer</button>
      </div>
    </form>
  );
}
