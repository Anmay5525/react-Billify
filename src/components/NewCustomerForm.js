import React from "react";
import PropTypes from "prop-types";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

export default function NewCustomerForm({ handleSubRouteChange }) {
  const handleNewCustomer = (e) => {
    e.preventDefault();
    const url = "https://rzp-training.herokuapp.com/team1/customers";

    const name = e.target.name.value;
    const email = e.target.email.value;
    const contact = e.target.phone.value;

    const data = {
      name,
      email,
      contact,
    };

    fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    })
      .then((res) => {
        if (res.ok) {
          return res.json();
        }
        return toast.error(
          `Something went wrong. Server responded with status ${res.status}`
        );
      })
      .then((r) => {
        if (r) {
          toast.success("New customer created");
          handleSubRouteChange("list");
        }
      })
      .catch((error) => console.log(error));
  };

  return (
    <div className="content">
      <form className="customers-form" onSubmit={(e) => handleNewCustomer(e)}>
        <div className="input-cnt">
          <div>Name</div>
          <input
            className="input"
            name="name"
            type="text"
            placeholder="Enter name of customer"
            required
            maxLength="25"
          />
        </div>

        <div className="input-cnt">
          <div>Email</div>
          <input
            className="input"
            type="email"
            name="email"
            placeholder="Enter email"
            required
            maxLength="30"
          />
        </div>

        <div className="input-cnt">
          <div>Phone No.</div>
          <input
            className="input"
            type="text"
            pattern="^[0-9]{10}$"
            name="phone"
            placeholder="Enter phone no."
            required
            onInput={(event) => {
              if (event.target.validity.patternMismatch) {
                event.target.setCustomValidity("Enter a valid phone number!");
              } else {
                event.target.setCustomValidity("");
              }
            }}
          />
        </div>
        <div>
          <button type="submit" className="customers-add-btn">
            Add Customer
          </button>
        </div>
      </form>
    </div>
  );
}

NewCustomerForm.propTypes = {
  handleSubRouteChange: PropTypes.func.isRequired,
};
