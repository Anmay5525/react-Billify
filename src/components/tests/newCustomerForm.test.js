import React from 'react';
import renderer from 'react-test-renderer';
import NewCustomerForm from '../NewCustomerForm';

it("newCustomerForm runs without crashing",()=>{
    const tree=renderer.create(<NewCustomerForm/>);
    expect(tree).toMatchSnapshot();
});