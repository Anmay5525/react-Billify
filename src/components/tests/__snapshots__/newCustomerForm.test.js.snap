// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newCustomerForm runs without crashing 1`] = `
<div
  className="content"
>
  <h2
    className="new-customer-title"
  >
    New Customer
  </h2>
  <div
    className="customers-form-container"
  >
    <form
      className="customers-form"
      onSubmit={[Function]}
    >
      <div
        className="input-cnt"
      >
        <div>
          Name
        </div>
        <input
          className="input"
          maxLength="25"
          name="name"
          placeholder="Enter name of customer"
          required={true}
          type="text"
        />
      </div>
      <div
        className="input-cnt"
      >
        <div>
          Email
        </div>
        <input
          className="input"
          maxLength="30"
          name="email"
          placeholder="Enter email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="input-cnt"
      >
        <div>
          Phone No.
        </div>
        <input
          className="input"
          name="phone"
          onInput={[Function]}
          pattern="^[0-9]{10}$"
          placeholder="Enter phone no."
          required={true}
          type="text"
        />
      </div>
      <div>
        <button
          className="customers-add-btn"
          type="submit"
        >
          Add Customer
        </button>
      </div>
    </form>
  </div>
</div>
`;
